tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/block/TsBlockBuilder.java

```diff
@@ -0,0 +1,295 @@
+    valueColumnBuilders = new ColumnBuilder[types.size()];
+    int initialExpectedEntries = timeColumnBuilder.getPositionCount();
+    for (int i = 0; i < valueColumnBuilders.length; i++) {
+      // TODO use Type interface to encapsulate createColumnBuilder to each concrete type class
+      // instead of switch-case
+      switch (types.get(i)) {
+        case BOOLEAN:
+          valueColumnBuilders[i] =
+              new BooleanColumnBuilder(
+                  tsBlockBuilderStatus.createColumnBuilderStatus(), initialExpectedEntries);
+          break;
+        case INT32:
+          valueColumnBuilders[i] =
+              new IntColumnBuilder(
+                  tsBlockBuilderStatus.createColumnBuilderStatus(), initialExpectedEntries);
+          break;
+        case INT64:
+          valueColumnBuilders[i] =
+              new LongColumnBuilder(
+                  tsBlockBuilderStatus.createColumnBuilderStatus(), initialExpectedEntries);
+          break;
+        case FLOAT:
+          valueColumnBuilders[i] =
+              new FloatColumnBuilder(
+                  tsBlockBuilderStatus.createColumnBuilderStatus(), initialExpectedEntries);
+          break;
+        case DOUBLE:
+          valueColumnBuilders[i] =
+              new DoubleColumnBuilder(
+                  tsBlockBuilderStatus.createColumnBuilderStatus(), initialExpectedEntries);
+          break;
+        case TEXT:
+          valueColumnBuilders[i] =
+              new BinaryColumnBuilder(
+                  tsBlockBuilderStatus.createColumnBuilderStatus(), initialExpectedEntries);
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown data type: " + types.get(i));
+      }
+    }
+  }
+
+  public void reset() {
+    if (isEmpty()) {
+      return;
+    }
+    tsBlockBuilderStatus =
+        new TsBlockBuilderStatus(tsBlockBuilderStatus.getMaxTsBlockSizeInBytes());
+
+    declaredPositions = 0;
+
+    for (int i = 0; i < valueColumnBuilders.length; i++) {
+      valueColumnBuilders[i] =
+          valueColumnBuilders[i].newColumnBuilderLike(
+              tsBlockBuilderStatus.createColumnBuilderStatus());
+    }
+  }
+
+  public TsBlockBuilder newTsBlockBuilderLike() {
+    return new TsBlockBuilder(
+        tsBlockBuilderStatus.getMaxTsBlockSizeInBytes(),
+        types,
+        timeColumnBuilder,
+        valueColumnBuilders);
+  }
+
+  public TimeColumnBuilder getTimeColumnBuilder() {
+    return timeColumnBuilder;
+  }
+
+  public ColumnBuilder getColumnBuilder(int channel) {
+    return valueColumnBuilders[channel];
+  }
+
+  public TSDataType getType(int channel) {
+    return types.get(channel);
+  }
+
+  public void declarePosition() {
+    declaredPositions++;
+  }
+
+  public void declarePositions(int positions) {
+    declaredPositions += positions;
+  }
+
+  public boolean isFull() {
+    return declaredPositions == Integer.MAX_VALUE || tsBlockBuilderStatus.isFull();
```

Comment:
Actually it's copied from `trino`, when I see this line of code, I've got the same confusion. Afterwards, I think it may just a kind of defensive programming because the max length for `array` in `java` is `Integer.MAX_VALUE`. Since we use array in ColumnBuilder to save data points, so it want to make sure that we won't break that rule.